File: src/main/java/dev/slne/protect/paper/command/commands/ProtectionWhoCommand.java
package dev.slne.protect.paper.command.commands;

import com.sk89q.worldguard.protection.regions.ProtectedRegion;
import dev.jorel.commandapi.CommandAPICommand;
import dev.slne.protect.paper.message.MessageManager;
import dev.slne.protect.paper.region.ProtectionUtils;
import dev.slne.protect.paper.region.info.RegionInfo;
import org.bukkit.Location;

import java.util.Set;

public class ProtectionWhoCommand extends CommandAPICommand {

	/**
	 * The {@link ProtectionWhoCommand}
	 */
	public ProtectionWhoCommand() {
		super("pwho");

		withPermission("surf.protect.pwho");

		executesPlayer((player, args) -> {
			Location playerLocation = player.getLocation();

			Set<ProtectedRegion> regions = ProtectionUtils.getProtectedRegionsByLocation(playerLocation);

			if (regions.isEmpty()) {
				player.sendMessage(MessageManager.getNoPlayerDefinedRegionComponent());
			}

			for (ProtectedRegion protectedRegion : regions) {
				RegionInfo regionInfo = new RegionInfo(player.getWorld(), protectedRegion);
				player.sendMessage(MessageManager.getPWhoComponent(regionInfo));
			}
		});
	}

}

File: src/main/kotlin/dev/slne/surf/protect/paper/command/commands/ProtectionWhoCommand.kt
package dev.slne.surf.protect.paper.command.commands

import dev.jorel.commandapi.kotlindsl.commandAPICommand
import dev.jorel.commandapi.kotlindsl.playerExecutor
import dev.slne.surf.protect.paper.message.Messages
import dev.slne.surf.protect.paper.permission.ProtectPermissionRegistry
import dev.slne.surf.protect.paper.region.info.RegionInfo
import dev.slne.surf.protect.paper.util.getProtectedRegions
import dev.slne.surf.surfapi.core.api.messages.adventure.sendText
import net.kyori.adventure.text.Component

fun protectionWhoCommand() = commandAPICommand("pwho") {
    withPermission(ProtectPermissionRegistry.PROTECTION_WHO_COMMAND)
    playerExecutor { player, _ ->
        val loc = player.location
        val regions = loc.getProtectedRegions()

        if (regions.isEmpty()) {
            player.sendText {
                appendErrorPrefix()
                error("Du stehst in keiner von einem Spieler gesicherten Region.")
            }
        }

        player.sendText {
            appendInfoPrefix()
            appendCollectionNewLine(regions, Component.empty()) { region ->
                Messages.Command.PWho.renderInfo(RegionInfo(region))
            }
        }
    }
}